### ISSUE TITLE
Fragment: null in SPAN Galaxy Plugin log

### ISSUE THREAD

**Issue by iromeo** [JetBrains-Research/galaxy-applications#9]
Fragment: null in SPAN Galaxy Plugin log
```
[Nov 26, 2018 10:08:50] FRAGMENT: null
```
Could we use more user-friendly values, e.g. show default value or "auto" or etc.

**Comment by dievsky** [comment]
Seems closely related to the "coverage upgrade" pull request in Epigenome, so I'm reassigning this to myself.

**Comment by dievsky** [comment]
Current (unreleased) version of SPAN prints messages like
```
Fragment size for [file].bam: 136 (cross-correlation estimate)
```
for single-end source files and nothing for paired-end.

**Comment by iromeo** [comment]
Could we show average fragment size or whatever for PE files? Just not to be a black box for a user? AFAIK MACS2 shows average fragment size for PE reads in it's log info.

**Comment by dievsky** [comment]
For paired-end sources, that's more of a statistic. We could print any statistic we want (the total number of reads, average coverage etc.), the question is: should we?

**Comment by iromeo** [comment]
I think it should be consistent with existing tools, like MAC2, SICER. I don't think that `total number of reads` or `average coverage` is a good option for fragment size value. We could show them in log as other tools do, but fragment size should be at least fragment size.

**Comment by dievsky** [comment]
The imputed fragment size for single-end sources and the average fragment size for the paired-end ones are completely different things. The first is a trained (or supplied) parameter used to process the data. The second one is just an immutable feature of the said data. If we report them in the same fashion, we'll trick users into thinking they can change fragment size for paired-end sources.

**Comment by iromeo** [comment]
I'd like not to confuse users where my "fragment" size section has gone. We could leave some information, like that fragment size inferred from mate pairs, or etc.

> users into thinking they can change fragment size for paired-end sources.
I thought users can always set manually any custom fragment size, no?

**Comment by dievsky** [comment]
The user-set fragment size is ignored for paired-end sources, because we _know_ the _exact_ fragment size for every read pair.
We can mention the single-endedness in the log next to the reported fragment size, like:
```
Fragment size for single-ended [file].bam: 136 (cross-correlation estimate)
```
and add something like
```
[file].bam is paired-end, ignoring fragment size.
```
for paired-end.

**Comment by iromeo** [comment]
SICER and MACS2 logs normally contain `KEY: VALUE` or ` KEY = VALUE` properties, so I suggest to use less text and more descriptive logging. E.g.

```
Library: Single-Ended
Fragment size: 136 bp (cross-correlation estimate)
```
or
```
Library: Pair-Ended
Fragment size: inferred from read pairs
```

`SICER`
```
GSM3074494_C57BL_6_oocyte_H3K36me3_rep1_mm10.bam: no control file
#############################################
######           SICER v1.1            ######
#############################################
Input library directory: /tmp/5495115.mgt2.cluster//GSM3074494_C57BL_6_oocyte_H3K36me3_rep1_mm10_F150_W200_G0_FDR0.01
ChIP library: GSM3074494_C57BL_6_oocyte_H3K36me3_rep1_mm10.bed
Output directory: /tmp/5495115.mgt2.cluster//GSM3074494_C57BL_6_oocyte_H3K36me3_rep1_mm10_F150_W200_G0_FDR0.01/out
Species: mm10
Threshold for redundancy allowed for reads: 1
Window size: 200 bps
Fragment size: 150 bps. The shift for reads is half of 150
Effective genome size as a fraction of the reference genome of mm10: 0.6847630188293125
Gap size: 0 bps
Evalue for identification of significant islands: 0.01


Preprocess the raw GSM3074494_C57BL_6_oocyte_H3K36me3_rep1_mm10 file to remove redundancy with threshold 1...
python2 /home/oshpynov/SICER_V1.1/SICER/src/remove_redundant_reads.py -s mm10 -b /tmp/5495115.mgt2.cluster//GSM3074494_C57BL_6_oocyte_H3K36me3_rep1_mm10_F150_W200_G0_FDR0.01/GSM3074494_C57BL_6_oocyte_H3K36me3_rep1_mm10.bed -t 1 -o /tmp/5495115.mgt2.cluster//GSM3074494_C57BL_6_oocyte_H3K36me3_rep1_mm10_F150_W200_G0_FDR0.01/out/GSM3074494_C57BL_6_oocyte_H3K36me3_rep1_mm10-1-removed.bed
chr1 	Plus reads: 2026844 	Retained plus reads: 1084986 ;	Minus reads: 2026844 	Retained minus reads: 1084330
chr2 	Plus reads: 2109304 	Retained plus reads: 998144 ;	Minus reads: 2109304 	Retained minus reads: 998197
...
```

`MACS2` 

```
--------------------------------------------------------------
Macs2 TMPDIR: /tmp/5478662.mgt2.cluster/
GSM3074495_Cast_Ei_oocyte_H3K36me3_rep1_mm10.bam: no control file
INFO  @ Fri, 12 Oct 2018 04:32:59:
# Command line: callpeak --tempdir /tmp/5478662.mgt2.cluster/ -t GSM3074495_Cast_Ei_oocyte_H3K36me3_rep1_mm10.bam -f BAM -g mm -n GSM3074495_Cast_Ei_oocyte_H3K36me3_rep1_mm10_broad_0.1 --broad --broad-cutoff 0.1 -B
# ARGUMENTS LIST:
# name = GSM3074495_Cast_Ei_oocyte_H3K36me3_rep1_mm10_broad_0.1
# format = BAM
# ChIP-seq file = ['GSM3074495_Cast_Ei_oocyte_H3K36me3_rep1_mm10.bam']
# control file = None
# effective genome size = 1.87e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff for narrow/strong regions = 5.00e-02
# qvalue cutoff for broad/weak regions = 1.00e-01
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 10000 bps
# Broad region calling is on
# Paired-End mode is off

INFO  @ Fri, 12 Oct 2018 04:32:59: #1 read tag files...
INFO  @ Fri, 12 Oct 2018 04:32:59: #1 read treatment tags...
INFO  @ Fri, 12 Oct 2018 04:33:18:  1000000
...
```

**Comment by iromeo** [comment]
> The user-set fragment size is ignored for paired-end sources
Maybe it is better to fail in this case? A user could never find some notice somewhere in the log file, that his cmdline attributes actually ignored and smth else is used instead. It is badly discoverable and a bit confusing behaviour...

**Comment by dievsky** [comment]
Your suggestion is more concise, I agree. I'll implement it.

**Comment by dievsky** [comment]
Done. Now we have:
1. SPAN logging its command-line arguments (`FRAGMENT: <fragment>` if provided, nothing otherwise)
2. ReadsQuery logging the library type and its thoughts on the fragment size.

I hope that now everything is close to perfect.

**Comment by olegs** [comment]
Fixed in SPAN 0.9.0
